flytekit/core/type_engine.py

```diff
@@ -757,6 +757,83 @@ def guess_python_type(self, literal_type: LiteralType) -> Type[T]:
         raise ValueError(f"Transformer {self} cannot reverse {literal_type}")
 
 
+class EnumTransformer(TypeTransformer[enum.Enum]):
```

Comment:
I didn't change this transformer, just had to move it up so that it can be used in TypeEngine.